tests/test_raw_node.rs

```diff
@@ -141,6 +141,56 @@ fn test_raw_node_propose_and_conf_change() {
     assert_eq!(entries[1].get_data(), &*ccdata);
 }
 
+// test_raw_node_read_index ensures that RawNode.read_index sends the MsgReadIndex message
+// to the underlying raft. It also ensures that ReadState can be read out.
+#[test]
+fn test_raw_node_read_index() {
```

Comment:
This test code is not a literal copy of the corresponding one in etcd/raft. In etcd/raft, the `raft.Step()` is a function pointer and could be easily replaced to ensure the message is delivered to the underlying raft. This test is done by checking the ready that is read out.